cogs/games.py

```diff
@@ -23,15 +23,174 @@
+            category = "country"
+        elif command in ["animals"]:
+            category = "animal"
+        else:
+            category: str = command or random.choice(
+                list(self.hangman_dict.keys()))
+        try:
+            word_list, pretty_name = self.hangman_dict[category]
+        except KeyError:
+            cat_list: str = ", ".join(
+                f"`{hm_cat}` (length: {len(self.hangman_dict[hm_cat][0])})"
+                for hm_cat in sorted(self.hangman_dict.keys()))
+            await ctx.send(f"command `{command}` is not valid\n"
+                           f"here is a list of valid commands: {cat_list}")
+            return
+
+        game_state = HangmanState(
+            "[REDACTED]" if command is None else pretty_name, word_list)
+        timeout_dict: dict[discord.Member, float] = {}
+        winner: Optional[discord.Member] = None
+        cool_win: bool = False
+
+        def wait_for_check(msg) -> bool:
+            return msg.channel == ctx.message.channel and (
+                (len(msg.content) == 1 and msg.content.isalpha())
+                or msg.content.lower() == game_state.lword)
+
+        await ctx.send(embed=game_state.embed)
+
+        while True:
+
+            try:
+                curr_msg = await self.bot.wait_for("message",
+                                                   check=wait_for_check,
+                                                   timeout=self.hm_timeout)
+            except asyncio.TimeoutError:
+                game_state.add_msg("the game has timed out")
+                await ctx.send(embed=game_state.embed)
+                await ctx.send(
+                    f"sorry everyone, no one has interacted with the "
+                    f"hangman in {self.hm_timeout//60} minutes, "
+                    f"the game has timed out")
+                return
+
+            curr_guess = curr_msg.content.lower()
+            if not (curr_msg.author in timeout_dict and
+                    (time() - timeout_dict[curr_msg.author]) < 1.5):
+                if curr_guess == game_state.lword:
+                    winner = curr_msg.author
+                    cool_win = game_state.full()
+                    if command is None:
+                        game_state.field_name = f"hangman (category: {pretty_name})"
+                    game_state.add_msg(f"{winner} guessed the entire word!")
+                    if game_state.img:
+                        game_state.embed.set_image(url=game_state.img)
+                    await ctx.send(embed=game_state.embed)
+                    await ctx.send(
+                        f"congratulations `{winner}`, you solved the hangman" +
+                        (f" (in a cool way), earning you {self.hm_cool_win} cheeps"
+                         if cool_win else
+                         f", earning you {self.hm_norm_win} cheeps"))
+                    break
+                if curr_guess in game_state.previous_guesses:
```

Comment:
more whitespace between blocks for all these big conditionals to improve readability